### PR TITLE
CVE product sections

### PR DESCRIPTION
**Pull request by chekl**

### CODE REVIEW

File: src/shared/types/ProductTypes.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
+export type ProductSectionUnion = "new" | "popular";
```
Comment (liubomyr123) [inline src/shared/types/ProductTypes.ts]: Are u going to use this type somewhere else ProductSection component? If not put it localy inside ProductSection.

File: src/shared/UI/ProductSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export default function ProductSection ({ section }: { section: ProductSectionUnion; }): JSX.Element {
+  const t = useTypedTranslation();
+  const { data, isLoading, error } = section === "new" ? useGetNewProductsQuery("") : useGetPopularProductsQuery("");
+  if (isLoading || !data || error) return <p>Loading...</p>;
+
+  return (
+    <section className="my-20 w-full">
+      <h5 className="h5 text-center">{section === "new" ? t("newProductSection") : t("popularProductSection")}</h5>
+      <div className="my-12 flex overflow-x-auto overflow-y-hidden px-10 before:m-auto after:m-auto">
```
Comment (liubomyr123) [inline src/shared/UI/ProductSection/index.tsx]: I think we do not need px-10. On small screens it looks not good.

Comment (liubomyr123) [inline src/shared/UI/ProductSection/index.tsx]: Let's use horisontal scroll and hide scrollbar. It will look good.

Comment (liubomyr123) [reply src/shared/UI/ProductSection/index.tsx]: Here is example i used when i work with pure js. U can use it with TS.
[useHorizontalScroll.txt](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/files/12516559/useHorizontalScroll.txt)

Comment (liubomyr123) [reply src/shared/UI/ProductSection/index.tsx]: [Screencast from 04.09.23 23:29:04.zip](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/files/12516571/Screencast.from.04.09.23.23.29.04.zip)

**Comment by liubomyr123** [comment]
Please move UpArrow component from footer to global Layout. So, on each page we still will see this arrow. And use useHideOnScroll for hiding arrow on scroll:
[useHideOnScroll.txt](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/files/12516654/useHideOnScroll.txt)

File: src/shared/UI/ProductSection/index.tsx (hunk 2 above)
Comment (chekl) [reply src/shared/UI/ProductSection/index.tsx]: I do the same but in CSS. I see that the scrollbar did not hide in firefox and I fixed it.

File: src/app/styles/normalize.scss [hunk 3]
```diff
@@ -24,21 +25,10 @@ code {
...
-        background-color: #a6b2c3;
-        border-radius: 5px;
-    }
-
-    &-corner {
-        background-color: #ffffff;
-    }
-}
\ No newline at end of file
+    width: 0;
```
Comment (liubomyr123) [inline src/app/styles/normalize.scss:28]: Please, do not change normilize styles. Create different file and there rewrite scrollbar.

**Comment by liubomyr123** [comment]
image
Please, check this case, when name can be too long.

File: src/shared/UI/ProductSection/index.tsx (hunk 2 above)
Comment (liubomyr123) [reply src/shared/UI/ProductSection/index.tsx]: [Screencast from 05.09.23 22:24:20.webm](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/assets/62620618/113bd491-8e1c-44f9-98d7-942b83aed9bb)
Okey, yes u hidded scrollbar, but now user can not scroll list(only if press 'shift' and then scroll). We can use hook or return back to old version? but add styles for chrome version of scroll. 

On video u can not see, but i try to scroll list with only mouse scroll.

Here is example with tailwind: https://dev.to/derick1530/how-to-create-scrollable-element-in-tailwind-without-a-scrollbar-4mbd

**Comment by liubomyr123** [comment]
[Screencast from 05.09.23 22:29:53.zip](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/files/12527694/Screencast.from.05.09.23.22.29.53.zip)
As u can see on video when we have small screen, UpArrow button will not help user to navigate to the beginning of page if it is in the end of page. So, i think it will be great do not attach it to footer but to screen itself. So in this case it does not meter if screen is big or small - we will always this button.

**Comment by liubomyr123** [comment]
[Screencast from 05.09.23 22:35:28.zip](https://github.com/Vladyslav-Samokryk/gift-hub-frontend/files/12527729/Screencast.from.05.09.23.22.35.28.zip)

Maybe we can change styles for footer a little bit later for mobile compatibility?
The same for header.

**Comment by liubomyr123** [comment]
Why u closed this PR?

**Comment by chekl** [comment]
> Why u closed this PR?

Because I added these changes to cve-random-present PR (without your comment about too long names in the product card, I think it is redundant)

**Pull request closed without merging**